interfaces/connection.go

```diff
@@ -45,6 +46,39 @@ type ConnectedSlot struct {
 	dynamicAttrs map[string]interface{}
 }
 
+// AttrGetter is an interface with Attr getter method common
+// to ConnectedSlot, ConnectedPlug, PlugInfo and SlotInfo types.
+type AttrGetter interface {
+	Attr(key string) (interface{}, error)
+}
+
+func getAttribute(staticAttrs map[string]interface{}, dynamicAttrs map[string]interface{}, key string, val interface{}) error {
+	var v interface{}
+	var ok bool
+
+	if dynamicAttrs != nil {
+		v, ok = dynamicAttrs[key]
+	}
+	if !ok && staticAttrs != nil {
+		v, ok = staticAttrs[key]
+	}
+
+	if ok {
```

Comment:
Instead of indenting everything and having to keep in mind what the condition was by the time we get to the last line, it's cleaner to invert the logic and repeat the above statement:
```
if !ok {
    return fmt.Errorf("attribute %q not found", key)
}
(... rest of the logic ...)
```

Reply:
This would benefit from context as well. See the related review point.